torch/jit/__init__.py

```diff
@@ -664,8 +664,8 @@ def trace(func,
           _force_outplace=False,
           _module_class=None):
     """
-    Trace a function and return an executable ``ScriptModule`` that will be optimized
-    using just-in-time compilation.
+    Trace a function and return an executable ``ScriptModule`` or ``torch.jit._C.Function``
```

Comment:
Well, if we ever want to mention this type then we should expose it as `torch.jit.Function`. The `_C` namespace should remain private forever, because it's an implementation detail.

Reply:
I'd rather have `trace` always return a `ScriptModule`, but @zdevito doesn't quite like this idea. So, yeah, we should probably have something user facing

Reply:
FWIW I agree with @zdevito on this one. I only didn't want us to mention the private namespaces in the docs.

Reply:
I'll put this as my TODO